core/src/main/java/org/jdbi/v3/core/qualifier/Qualifiers.java

```diff
@@ -30,25 +29,32 @@
  * Utility class for type qualifiers supported by Jdbi core.
  */
 @Beta
-public class Qualifiers {
-    private static final ExpiringMap<Object, Set<Annotation>> ANNOS = ExpiringMap.builder()
-            .expiration(10, TimeUnit.MINUTES)
-            .build();
+public class Qualifiers extends JdbiCache<Qualifiers, Object, Set<Annotation>> {
+    public Qualifiers() {}
 
-    private Qualifiers() {}
+    /**
+     * Returns the set of qualifying annotations on the given elements.
+     * @param elements the annotated elements. Null elements are ignored.
+     * @return the set of qualifying annotations on the given elements.
+     */
+    public Set<Annotation> qualifiers(AnnotatedElement... elements) {
```

Comment:
This is admittedly a little weird, but `Qualifiers` holds on to lots of reflection objects, which is a source of ClassLoader retention.  I made the cached version instance rather than static (and left the static method "just in case", e.g. in the SqlObject handler creation it hardly matters since those get cached anyway)